pkgs/build-support/fetchgithub/default.nix

```diff
@@ -53,16 +61,19 @@ let
   fetcherArgs = (if useFetchGit
     then {
       inherit rev deepClone fetchSubmodules sparseCheckout; url = gitRepoUrl;
+      passthru = newPassthru;
     } // lib.optionalAttrs (leaveDotGit != null) { inherit leaveDotGit; }
     else {
       url = "${baseUrl}/archive/${rev}.tar.gz";
 
-      passthru = {
+      passthru = newPassthru // {
         inherit gitRepoUrl;
       };
     }
   ) // privateAttrs // passthruAttrs // { inherit name; };
 in
 
-fetcher fetcherArgs // { meta = newMeta; inherit rev owner repo; }
+(fetcher fetcherArgs).overrideAttrs (finalAttrs: previousAttrs: {
```

Comment:
Any specific reason for `overrideAttrs` over `//`?

Reply:
It's not super clear to me. The original author explains it as:
> Enhance overriding for result FOD of fetchFromGitHub with <pkg>.overrideAttrs and passthru.
>  - Update the meta attribute of the result package by <pkgs>.overrideAttrs the one returned by the base fetcher.
>  - Attach owner, repo and rev attributes via passthru instead of directly updating the attribute set via //, preserving these attributes across <pkg>.overrideAttrs.

https://github.com/NixOS/nixpkgs/pull/294329

It's true that if someone re-overrides after the `//`, the meta will be lost. However, since `fetchFromGitHub` overrides `overrideAttrs` anyway (with `makeOverridable`) I don't think this can happen in practice. Better safe than sorry?

Reply:
Mhmm. Sounds good to me.